store/src/state.rs

```diff
@@ -176,7 +189,7 @@ impl From<NullifierStateForTransactionInput> for NullifierTransactionInputRecord
 impl State {
     /// Loads the state from the `db`.
     #[instrument(target = "miden-store", skip_all)]
-    pub async fn load(mut db: Db) -> Result<Self, anyhow::Error> {
+    pub async fn load(mut db: Db) -> Result<Self> {
```

Comment:
I would probably create a special error type for this function. Something like: `StateInitializationError`.